packages/assets-controllers/src/CurrencyRateController.ts

```diff
@@ -1,33 +1,36 @@
 import type { RestrictedControllerMessenger } from '@metamask/base-controller';
-import { BaseControllerV2 } from '@metamask/base-controller';
 import {
   TESTNET_TICKER_SYMBOLS,
   FALL_BACK_VS_CURRENCY,
-  safelyExecute,
 } from '@metamask/controller-utils';
+import type {
+  NetworkClientId,
+  NetworkControllerGetNetworkClientByIdAction,
+} from '@metamask/network-controller';
+import { PollingController } from '@metamask/polling-controller';
 import { Mutex } from 'async-mutex';
 import type { Patch } from 'immer';
 
 import { fetchExchangeRate as defaultFetchExchangeRate } from './crypto-compare';
 
 /**
  * @type CurrencyRateState
- * @property conversionDate - Timestamp of conversion rate expressed in ms since UNIX epoch
- * @property conversionRate - Conversion rate from current base asset to the current currency
+ * @property currencyRates - Object keyed by native currency
+ * @property currencyRates.conversionDate - Timestamp of conversion rate expressed in ms since UNIX epoch
+ * @property currencyRates.conversionRate - Conversion rate from current base asset to the current currency
  * @property currentCurrency - Currently-active ISO 4217 currency code
- * @property nativeCurrency - Symbol for the base asset used for conversion
- * @property pendingCurrentCurrency - The currency being switched to
- * @property pendingNativeCurrency - The base asset currency being switched to
  * @property usdConversionRate - Conversion rate from usd to the current currency
  */
 export type CurrencyRateState = {
-  conversionDate: number | null;
-  conversionRate: number | null;
   currentCurrency: string;
-  nativeCurrency: string;
-  pendingCurrentCurrency: string | null;
-  pendingNativeCurrency: string | null;
-  usdConversionRate: number | null;
+  currencyRates: Record<
+    string,
```

Comment:
Should we use an alias for the string, so that the type is self describing? Maybe something like the following:

```
// docs about what is native currency
type NativeCurrency = string;

...

    Record<
      NativeCurrency,
      {
         ...
```